insights/specs/datasources/cloud_init.py

```diff
@@ -0,0 +1,94 @@
+"""
+Custom datasources for cloud initialization information
+"""
+import json
+import yaml
+
+from insights.core.context import HostContext
+from insights.core.dr import SkipComponent
+from insights.core.plugins import datasource
+from insights.core.spec_factory import DatasourceProvider, simple_file
+from insights.specs import Specs
+
+
+class LocalSpecs(Specs):
+    """ Local specs used only by cloud_init datasources """
+
+    cloud_cfg_input = simple_file("/etc/cloud/cloud.cfg")
+    """ Returns the contents of the file ``/etc/cloud/cloud.cfg`` """
+
+
+@datasource(LocalSpecs.cloud_cfg_input, HostContext)
+def cloud_cfg(broker):
```

Comment:
@bfahr, The datasource `cloud_cfg` should be in the class `LocalSpecs`, or else insights.specs.Specs.cloud_cfg can't find the dependency, right?

Reply:
@huali027 `cloud_cfg` that is defined in default.py is the data that will be collected in the archive.  `cloud_cfg_input` is only collected for the datasource processing, and is not stored in the archive.  Its defined locally in the same module to simplify the implementation.

Reply:
@bfahr, got it, I found "cloud_cfg = cloud_init.cloud_cfg" in default.py, thanks.